### ISSUE TITLE
Search highlighting makes text within Mermaid-rendered SVG diagrams disappear when matched

### ISSUE THREAD

**Issue by aral** [vjeantet/hugo-theme-docdock#112]
![search-bug](https://user-images.githubusercontent.com/41798/34355203-dace6544-ea33-11e7-8892-ff5d55824e81.png)

Search currently marks up text within SVG diagrams using the `<mark>` tag, which results in the text disappearing (the opposite of what we want) :) 

There seems to be an attempt to avoid this: https://github.com/vjeantet/hugo-theme-docdock/blob/84b60f185a7b8d43b182f993c5f4501740a4aa38/static/js/docdock.js#L242  

But that doesn’t work, as the owner node of the text content (the text node) – at least in this sequence diagram – doesn’t have a class of ‘mermaid’:

<img width="906" alt="screen shot 2017-12-26 at 12 07 15" src="https://user-images.githubusercontent.com/41798/34355419-716bae02-ea35-11e7-8908-d09c1fc890fc.png">

Suggested solution:

Check if the parent node’s ownerSVGElement is an SVG element.

```javascript
if (match && !(node.parentNode.ownerSVGElement instanceof SVGElement)) {
```

Happy to send in a pull request if you agree, @vjeantet.

**Comment by aral** [comment]
I just had a quick look at the original Hugo Learn Theme and it doesn’t suffer from the same issue. Looks like they’re specifically handling this use case:

<img width="509" alt="screen shot 2017-12-26 at 12 16 34" src="https://user-images.githubusercontent.com/41798/34355585-d5337f2c-ea36-11e7-9182-c4dba625e613.png">

Instead of the workaround, above, this should be ported in.

**Comment by aral** [comment]
Scratch the previous comment, looks like an inconsistency in Mermaid. The above SVG is what it renders for a flowchart. For a sequence diagram, it doesn’t wrap the text in a foreign object tag and that’s what’s resulting in the behaviour I saw above. So I suggest we do implement the workaround I suggested in this issue and also inform both Hugo Learn (which suffers from the same issue) and Mermaid (where the upstream fix should originate, as far as I can see).
